fairseq2n/src/fairseq2n/data/concat_data_source.cc

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Meta Platforms, Inc. and affiliates.
+// All rights reserved.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+
+#include "fairseq2n/data/concat_data_source.h"
+#include <vector>
+
+namespace fairseq2n::detail {
+
+concat_data_source::concat_data_source(
+    std::vector<data_pipeline> &&pipelines)
+    : pipelines_{std::move(pipelines)}
```

Comment:
There is a subtle bug here. This is one of the complicated features of C++, but here you should use parentheses instead of brace initialization. Otherwise, the compiler will pick the initializer_list constructor instead of move constructor.

```cpp
: pipelines_(std::move(pipelines))
```

Reply:
Fixed